src/main/java/org/broadinstitute/hellbender/tools/spark/sv/GatherSplitReadsSpark.java

```diff
@@ -0,0 +1,664 @@
+        }
+
+        @Override
+        public GATKRead next()
+        {
+            if ( !hasNext() )
+                throw new NoSuchElementException("Iterator<GATKRead> is exhausted.");
+            return outputIterator.next();
+        }
+
+        @Override
+        public Iterator<GATKRead> iterator() { return this; }
+
+        private Iterator<GATKRead> processRead( final GATKRead read )
+        {
+            if ( !read.isUnmapped() &&
+                    read.getMappingQuality() >= MIN_MAPQ &&
+                    read.getStart() != ReadConstants.UNSET_POSITION )
+            {
+                String readContig = read.getContig();
+                if ( currentContig != readContig )
+                {
+                    currentContig = readContig;
+                    locMap.clear();
+                }
+
+                final List<CigarElement> cigarElements = read.getCigar().getCigarElements();
+                int matchLen = 0;
+                for ( CigarElement ele : cigarElements )
+                    if (ele.getOperator() == CigarOperator.MATCH_OR_MISMATCH)
+                        matchLen += ele.getLength();
+                if ( matchLen >= MIN_MATCH_LEN )
+                {
+                    EventLocus eventLocus = splitReadDetector.getLocusIfReadIsSplit(read);
+                    if ( eventLocus != null )
+                        return cluster(eventLocus,read);
+                    eventLocus = discordantPairDetector.getLocusIfDiscordant(read);
+                    if ( eventLocus != null )
+                        return cluster(eventLocus,read);
+                }
+            }
+            return Collections.emptyIterator();
+        }
+
+        private Iterator<GATKRead> cluster( final EventLocus newEventLocus, final GATKRead read )
+        {
+            final int locusStart = newEventLocus.getLocusStart();
+            final int locusEnd = newEventLocus.getLocusEnd();
+
+            // clean out old stuff that can't possibly be interesting anymore
+            final Iterator<Map.Entry<EventLocus,GATKRead>> itr = locMap.entrySet().iterator();
+            final int staleEnd = locusStart - MAX_LOCUS_DIST;
+            while ( itr.hasNext() )
+            {
+                final EventLocus eventLocus = itr.next().getKey();
+                if ( eventLocus.getLocusStart() >= locusStart )
+                    break;
+                if ( eventLocus.getLocusEnd() <= staleEnd )
+                    itr.remove();
+            }
+
+            locMap.put(newEventLocus.makeUnique(++eventIdCounter),read);
+
+            // see if there are a sufficient number of overlapping putative events to call a cluster
+            final List<Map.Entry<EventLocus,GATKRead>> overlappers = new LinkedList<>();
+            for ( Map.Entry<EventLocus,GATKRead> entry : locMap.entrySet() )
+            {
+                final EventLocus eventLocus = entry.getKey();
+                if ( eventLocus.getLocusStart() >= locusEnd )
```

Comment:
i need to know more than whether or not it overlaps:  i need to know relative placement of a disjoint region so i know when to break.  i could write a method that returned an enumeration { LEFT_DISJOINT, OVERLAPPING, RIGHT_DISJOINT }, i suppose, but it would be a lot more code.  i'm inclined to let the economy of the existing code rule, in this case.